sound/soc/sof/topology.c

```diff
@@ -1858,13 +1873,77 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 		offset += pdata->size;
 	}
 
+process:
 	process->size = ipc_data_size;
 	swidget->private = process;
 
 	ret = sof_ipc_tx_message(sdev->ipc, process->comp.hdr.cmd, process,
 				 ipc_size, r, sizeof(*r));
-	if (ret >= 0)
+
+	if (ret >= 0 && ipc_data_size)
 		return ret;
+
+	/* send control data with large message supported method */
+	for (i = 0; i < widget->num_kcontrols; i++) {
+		kc = &widget->kcontrol_news[i];
+
+		/* we only care about bytes control */
```

Comment:
you should elaborate on this, the comment is just too short to understand what you explained in the previous review (it doesn't make sense to send the other types through topology).

Reply:
actually this is now just wrong, will remove.